framework/deproxy_manager.py

```diff
@@ -13,26 +13,31 @@
 def finish_all_deproxy():
     asyncore.close_all()
 
-def run_deproxy_server(deproxy, exit_event, polling_lock):
+def run_deproxy_server(deproxy, exit_event, polling_lock, q):
     tf_cfg.dbg(3, "Running deproxy server manager")
 
-    if hasattr(select, 'poll'):
-        poll_fun = asyncore.poll2
-    else:
-        poll_fun = asyncore.poll
-    while not exit_event.is_set():
-        polling_lock.acquire()
-        poll_fun()
+    try:
+        if hasattr(select, 'poll'):
+            poll_fun = asyncore.poll2
+        else:
+            poll_fun = asyncore.poll
+        while not exit_event.is_set():
+            polling_lock.acquire()
+            poll_fun()
+            polling_lock.release()
+    except Exception as e:
+        tf_cfg.dbg(2, "Error while polling: %s" % str(e))
         polling_lock.release()
-
-    tf_cfg.dbg(3, "Stopped deproxy manager")
+        q.put(e)
+    tf_cfg.dbg(3, "Finished deproxy manager")
 
 class DeproxyManager(stateful.Stateful):
     """ Class for running and managing
     deproxy servers and clients. polling cycle is also here.
     Tests don't need to manually use this class."""
 
     def __init__(self):
+        self.exception = Queue.Queue()
```

Comment:
The Queue class implements thread-safe queue , so `safe.exception` is a little bit misleading name.